### ISSUE TITLE
Add dependabot

### ISSUE THREAD

**Issue by vladyslavstartsev** [phpmyadmin/phpmyadmin#16206]
### Describe the solution you'd like

I see that you make quite a lot of `composer.json` changes manually. Most of it is update of dependencies. 
What can automate this, is `dependabot`

It's easy to install it, you only need one file to configure (there is the example commit https://github.com/weirdan/codeception-psalm-module/commit/11d5faaa17b1e9f34a1e2cd0514bbbd9255fd90d)

Here are the examples 
https://github.com/weirdan/codeception-psalm-module/commit/11d5faaa17b1e9f34a1e2cd0514bbbd9255fd90d
https://github.com/doctrine/coding-standard/commit/feef039e4c314baab969c5391c9061c728548322

this way you can automate part of your job

**Comment by mauriciofauth** [comment]
phpMyAdmin doesn't have the `composer.lock` file in the repository, so I don't think _dependabot_ will help much in this situation.

**Comment by ibennetch** [comment]
Well, the https://dependabot.com/ page is quite light on details, but I would think it has to work with composer.json, right? Otherwise it wouldn't be too helpful for folks like us.

Anyway, I'm on the fence about this. I do like the idea, but find the documentation lacking and wonder how it would fit in to our development cycle. I wasn't able to find much information about how compatibility are resolved or whether major version changes are suggested.

It seems like a handy tool to decrease our time managing these dependencies, and honestly at first I was pretty enthusiastic about it, but reading the documentation and main web site leave me with more questions than I started with. That makes me hesitant and now I'm not sure what to think.

🤷‍♂️

**Comment by vladyslavstartsev** [comment]
@mauriciofauth 
It doesn't work with `composer.lock` (I mean, it can, but it's not necessary) see https://github.com/doctrine/coding-standard for example, `composer.lock` is in .gitignore.
As docs says, it's really flexible and powerful.
@ibennetch You asked me about major/minor versioning and how this is solved - https://help.github.com/en/github/administering-a-repository/configuration-options-for-dependency-updates#versioning-strategy . So it can update `composer.lock` if you want, or it can update `composer.json`. It's up to you how to configure it.
Here is how the PR looks like https://github.com/doctrine/coding-standard/pull/198

**Comment by williamdes** [comment]
Hi
I will just add my thoughts on dependabot
I have it on multiple organizations and it opens a lot of PRs very often
A pain in the * to use.
I use it to know there is a dependency to bump and then I read the changelog the bot posts. 
Next step I bump everything and push only one commit to close everything. 

My problem with dependabot: one PR for each dependency to bump

And trust me there is always someone that releases a version of something. 

TLDR; I am sorry I do not want dependabot, it is a nightmare of notifications and PRs.
Sure I am a perfectionist for GIT history and everything.. 

I do not want to have more notifications that I already have to manage ~1-2K emails per month. 

One day I will build a better BOT, and remove dependabot from all my repos. 

Anyway, to not take this personally I appreciate that you bring the tool up to us.

**Comment by vladyslavstartsev** [comment]
@williamdes thank you for your feedback.
I see your opinion 😉. 

Please close if there will be no more progress on this

**Comment by williamdes** [comment]
Thank you for your comprehension, I really appreciate ;)
I am closing this one for now

Have a nice day !
